addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -1028,4 +1050,29 @@ private ServerConnectionEstablisher getServerEstablisher() {
         }
         return serverEstablisher;
     }
+
+    private String getStatsHostname() {
+        if (statsHostName == null) {
+            // Make our best attempt at getting the same host name that other stats will use
+            HistoryReference hsr = getHandshakeReference();
+            if (hsr != null) {
+                statsHostName = getSiteName(host, hsr.getURI().getPort());
+            } else {
+                statsHostName = "http://" + host;
+            }
+        }
+        return statsHostName;
+    }
+
+    private static String getSiteName(String host, int port) {
```

Comment:
Should this use ws and wss, same for the method above?

Reply:
Well, then we get stats for the same logical site split over 2 different stats domains, which I didnt think was very useful. If I'm accessing https://www.example.com and that uses websockets then I'd like to be able to get all of the stats via https://www.example.com otherwise I have to put in a load of extra logic into my client and potentially change this everytime more stats are added.

Reply:
But I am thinking of my usecase where I'm scanning 100s of sites, so other views appreciated.

Reply:
Maybe it shouldn’t include scheme then?

Reply:
Well, we already include these for the http(s) urls, so I was fitting in with that :/

Reply:
Alright, it shouldn't be a blocker. Let's proceed like this, if people ask for something else we can always re-adapt it later.